src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +87,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()){
```

Comment:
Coding standard violation regarding spacing between braces and `if`. Refer to the coding standards guide for the solution.